Mathlib/RingTheory/RingHomProperties.lean

```diff
@@ -167,6 +165,45 @@ theorem StableUnderBaseChange.pushout_inl (hP : RingHom.StableUnderBaseChange @P
   apply hP R T S (TensorProduct R S T)
   exact H
 
+variable (hP : RingHom.StableUnderBaseChange @P)
+
+section
+
+variable {R S Rᵣ Sᵣ : Type u} [CommRing R] [CommRing S] [CommRing Rᵣ] [CommRing Sᵣ] [Algebra R Rᵣ]
+  [Algebra R S] [Algebra S Sᵣ] [Algebra R Sᵣ] [Algebra Rᵣ Sᵣ] [IsScalarTower R S Sᵣ]
+  [IsScalarTower R Rᵣ Sᵣ]
+
+include hP in
+/-- Let `S` be an `R`-algebra and `Sᵣ` and `Rᵣ` be the respective localizations at a submonoid
+`M` of `R`. If `P` is stable under base change and `P` holds for `algebraMap R S`, then
+`P` holds for `algebraMap Rᵣ Sᵣ`. -/
+lemma StableUnderBaseChange.of_isLocalization
+    (M : Submonoid R) [IsLocalization M Rᵣ] [IsLocalization (Algebra.algebraMapSubmonoid S M) Sᵣ]
+    (h : P (algebraMap R S)) : P (algebraMap Rᵣ Sᵣ) :=
+  letI : Algebra.IsPushout R S Rᵣ Sᵣ := Algebra.isPushout_of_isLocalization M Rᵣ S Sᵣ
+  hP R S Rᵣ Sᵣ h
+
+end
+
+variable {R S Rᵣ Sᵣ : Type u} [CommRing R] [CommRing S] [CommRing Rᵣ] [CommRing Sᵣ] [Algebra R Rᵣ]
+  [Algebra S Sᵣ]
+
+include hP in
+/-- If `P` is stable under base change and holds for `f`, then `P` holds for `f` localized
+at any submonoid `M` of `R`. -/
+lemma StableUnderBaseChange.isLocalization_map (M : Submonoid R) [IsLocalization M Rᵣ]
+    (f : R →+* S) [IsLocalization (M.map f) Sᵣ] (hf : P f) :
+    P (IsLocalization.map Sᵣ f M.le_comap_map : Rᵣ →+* Sᵣ) := by
+  letI : Algebra Rᵣ Sᵣ := (IsLocalization.map Sᵣ f M.le_comap_map).toAlgebra
+  letI : Algebra R S := f.toAlgebra
+  letI : Algebra R Sᵣ := ((algebraMap Rᵣ Sᵣ).comp (algebraMap R Rᵣ)).toAlgebra
+  haveI : IsScalarTower R S Sᵣ := IsScalarTower.of_algebraMap_eq'
+    (IsLocalization.map_comp M.le_comap_map)
+  haveI : IsScalarTower R Rᵣ Sᵣ := IsScalarTower.of_algebraMap_eq' rfl
+  haveI : IsLocalization (Algebra.algebraMapSubmonoid S M) Sᵣ :=
+    inferInstanceAs <| IsLocalization (M.map f) Sᵣ
```

Comment:
Can this be golfed with `algebraize`?

Reply:
Yes, and I had already done this in some other PR :D I had messed up the merging. Is now fixed.